INTEGRATION.md

```diff
@@ -0,0 +1,74 @@
+## Add "redhat.vscode-commons" as extension dependency in package.json file for auto download of vscode-commons
+
+- Dependencies to other extensions. The identifier of an extension is always ${publisher}.${name}. For example: vscode.csharp.
+
+```
+  "extensionDependencies": ["redhat.vscode-commons"],
+```
+
+## Add your custom segment key in package.json file
+
+- This key will be used to connect and push usage data to segment
+
+```
+    "segmentWriteKey": "your-segment-key-goes-here",
+```
+
+> **NOTE** Default segment key will be used if you do not provide a custom segment key
+
+## Add the bellow code in your client extensions
+
+```
+interface TelemetryEvent {
+  uuid?: string;
+  type?: string; // type of telemetry event such as : identify, track, page, etc.
+  name?: string;
+  properties?: any;
+  measures?: any;
+}
+
+async function telemetry(context: vscode.ExtensionContext) {
+  // To get an instance of "redhat.vscode-commons"
+  const vscodeCommons = vscode.extensions.getExtension("redhat.vscode-commons");
+  let vscodeCommonsIsAlive = false;
+
+  if (vscodeCommons?.isActive) {
+    console.log("redhat.vscode-commons is active");
+    vscodeCommonsIsAlive = true;
+  } else {
+    console.log("redhat.vscode-commons is not active");
+    await vscodeCommons?.activate().then(
+      function () {
+        console.log("redhat.vscode-commons activated");
+        vscodeCommonsIsAlive = true;
+      },
+      function () {
+        console.log("redhat.vscode-commons activation failed");
+      }
+    );
+  }
+
+  if (vscodeCommonsIsAlive) {
+    const extensionIdentifier = "redhat.alice";
+    const vscodeCommonsAPI = vscodeCommons?.exports;
+
+    /*
+    A "MUST HAVE" CALL
+    set segment key in package.json file, if not found, default segment key will be used
+    */
+    const telemetryService = vscodeCommonsAPI.getTelemetryService(
+      extensionIdentifier
+    );
+    context.subscriptions.push(telemetryService);
+
+    if (telemetryService) {
+      let event: TelemetryEvent = {
+        type: "track",
+        uuid: "redhat123321",
```

Comment:
Ya, we can pass it while creating the instance. This will be same for all cases.

Reply:
@mathur07 uuid should be set internally by the telemetry service (that's why we have a getRedHatUUID() method to implement). It makes no sense to allow different uuids from different extensions, since we would not be able to cross-reference users from different extensions

Reply:
@fbricon maybe we can leave this to the client/adopters extensions, so that client's UUID creation, deletion, and invalidation logic could persist in the client's hands and be tracked.
So we can have a logic that if the UUID is passed from the client we can use it else we can use getRedHatUUID() 
WDYT?

Reply:
I don't think we need/want extensions to control the creation of their own UUID  (remember these are only for Red Hat extensions, not other adopters). 
On the other hand, some extensions need to access the UUID value, that's why we discussed about exposing the getRedHatUUID() publicly for others to leverage.